cyences_app_for_splunk/bin/cyences_send_digest_email.py

```diff
@@ -13,113 +12,168 @@
-    severity = Option(name='cyences_severity', require=False, default=None)
-    results_link = Option(name="results_link", require=False, default=None)
-    trigger_time = Option(name="trigger_time", require=False, default=None)
-    results_file = Option(name="results_file", require=False, default=None)
+# We'll be sorting this in ascending order
+ALERT_SEVERITIES = {
+    'critical': 1,
+    'high': 2,
+    'medium': 3,
+    'low': 4,
+    'info': 5
+}
 
 
-    def check_session_key(self, records):
-        if not self.search_results_info or not self.search_results_info.auth_token:
-            logger.debug("Unable to find session key in the custom command. Logging records, if any.")
-            for r in records:
-                logger.debug(r)
-            raise Exception("unable to find session key.")
 
+@Configuration()
+class CyencesSendDigestEmailCommand(EventingCommand):
 
-    def results_by_alert(self, results, severity_str, exclude_alert_str):
+    alert_name = Option(name="alert_name", require=True)
+    email_to = Option(name='email_to', require=False, default=None)
+    severities = Option(name='cyences_severities', require=False, default=None)
+    max_results_per_alert = Option(name='max_results_per_alert', require=False, default=15, validate=validators.Integer(minimum=1, maximum=500))
+    max_alerts_per_email = Option(name='max_alerts_per_email', require=False, default=10, validate=validators.Integer(minimum=1, maximum=500))
 
-        severity_filter = cs_utils.convert_to_set(severity_str)
-        exclude_alert = cs_utils.convert_to_set(exclude_alert_str)
 
+    def filter_results_and_group_by_alert(self, results, severity_filter, alerts_to_exclude):
         logger.debug("severity_filter={}".format(severity_filter))
-        logger.debug("exclude_alert={}".format(exclude_alert))
+        logger.debug("alerts_to_exclude={}".format(alerts_to_exclude))
+
         alerts = {}
 
         for event in results:
-            alert_name = event['alert_name']
+            # logger.debug("Event: {}".format(event))
+            alert_name = event[FIELD_FOR_ALERT_NAME_IN_NOTABLE_EVENTS]
 
             # Skip event if alert_name in the exclude_alert
-            if alert_name.lower() in exclude_alert:
+            if alert_name.lower() in alerts_to_exclude:
                 continue
 
-            # skip event if and cyences_severity is not matching with severity_filter
-            if event.get('cyences_severity', '').lower() not in severity_filter:
-                continue
+            if 'cyences_severity' in event:
+                severity = event.get('cyences_severity', '').lower()
+
+                # Skip the event if cyences_severity is defined and not present in the severity_filter
+                if severity not in severity_filter:
+                    continue
+
+                try:
+                    event['__cyences_severity'] = ALERT_SEVERITIES[event['cyences_severity']]  # Add addition information for sorting
```

Comment:
should we use `severity` instead of `event['cyences_severity']` to make it case insensitive.